### PR TITLE
Crypten Plan Notebook - Training Encrypted NN

### PR DESCRIPTION
**Pull request by gmuraru**
## Description
A tutorial notebook for CrypTen using Plans.
This needs to be merged after the plan support PR is merged

- [x] - reorganize the folder structure from plans
- [x] - add documentation

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

**ATTENTION**:
This should be merged after [this](https://github.com/OpenMined/PySyft/pull/3531)

### CODE REVIEW

File: syft/execution/placeholder.py [hunk 1]
```diff
@@ -288,6 +288,19 @@ def extract(tensor):
             current_level = getattr(current_level, "child", None)
         return current_level
 
+    @staticmethod
+    def recursive_extract(results):
+        current_level = results
```
Comment (LaRiffle) [inline syft/execution/placeholder.py:293]: This var is defined but not used

File: examples/experimental/CrypTen/CrypTen - Traing Encrypted Neural Network - Plans.ipynb [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define the CrypTen computation\n",
+    "\n",
+    "We need to specify for the ```run_multiworkers``` decorater:\n",
+    "* the workers that will take part in the computation\n",
+    "* the master address, this will be used for their syincronization\n",
+    "* the insntiated model that will be sent\n",
+    "* a dummy input for the model\n",
```
Comment (LaRiffle) [inline examples/experimental/CrypTen/CrypTen - Traing Encrypted Neural Network - Plans.ipynb:267]:
```suggestion
    "We need to specify for the ```run_multiworkers``` decorator:\n",
    "* the workers that will take part in the computation\n",
    "* the master address, this will be used for their synchronization\n",
    "* the instantiated model that will be sent\n",
    "* a dummy input for the model\n",
```

File: examples/experimental/CrypTen/CrypTen - Traing Encrypted Neural Network - Plans.ipynb [hunk 3]
```diff
@@ -0,0 +1,543 @@
...
+    "* Install PySyft from the [`crypten` branch](https://github.com/OpenMined/PySyft/tree/crypten).\n",
+    "* Clone the [GridNode repository](https://github.com/OpenMined/GridNode)\n",
+    "  * we need this because *alice* and *bob* are two different GridNodes\n",
+    "\n",
+    "### Run the grid nodes\n",
+    "* In two separate terminals run:\n",
+    "    * ```python -m gridnode --id alice --port 3000```\n",
+    "    * ```python -m gridnode --id bob --port 30001```\n",
+    "    \n",
+    "This will start two workers, *alice* and *bob* and we will connect to them using the port 3000 and 30001.\n",
```
Comment (LaRiffle) [inline examples/experimental/CrypTen/CrypTen - Traing Encrypted Neural Network - Plans.ipynb:45]:
```suggestion
    "    * ```python -m gridnode --id bob --port 3001```\n",
    "    \n",
    "This will start two workers, *alice* and *bob* and we will connect to them using the port 3000 and 3001.\n",
```

File: syft/frameworks/crypten/hook/hook.py [hunk 4]
```diff
@@ -27,12 +76,27 @@ def f_return_model_or_cryptensor(*args, **kwargs):
         """
         return crypten.cryptensor(th.zeros([]))
 
+    @staticmethod
+    def f_return_module(*args, **kwargs):
+        return crypten.nn.Module()
+
+    @staticmethod
+    def f_return_local_worker(*args, **kwargs):
```
Comment (gmuraru) [inline syft/frameworks/crypten/hook/hook.py:84]: TODO: Remove this.

File: syft/frameworks/crypten/hook/hook.py [hunk 5]
```diff
@@ -135,6 +131,7 @@ def define_crypten_plan_hook():
         crypten: {
             "load": CrypTenPlanBuild.f_return_model_or_cryptensor,
             "cat": CrypTenPlanBuild.f_return_cryptensor,
+            "load_model": CrypTenPlanBuild.f_return_mode,
```
Comment (LaRiffle) [inline syft/frameworks/crypten/hook/hook.py:134]: typo
is it f_return_model ?

Comment (gmuraru) [reply syft/frameworks/crypten/hook/hook.py:134]: Yep...
